### PR TITLE
dash/datagrid-renamed-classes

### PR DESCRIPTION
**Pull request by sebastianbochan**
-  [x] - renaming all classes
-  [x] - rename head class
-  [x] - extract abstract Cell class and add separate classes for
   -  [x] `headerCell`
   -  [x] - row
   -  [x]  `tableCell`
-  [x] - add placeholder for footer class

### CODE REVIEW

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
When runnign tests i get:
```bash
[08:43:30] Linting TypeScript declarations (.d.ts) for dashboards ...
code/dashboards/es-modules/Dashboards/Plugins/DataGridTypes.d.ts(4,32): error TS2694: Namespace '"/Users/karolkolodziej/repos/highcharts/code/datagrid/datagrid.src"' has no exported member 'Column'.
code/dashboards/es-modules/Dashboards/Plugins/DataGridTypes.d.ts(5,29): error TS2694: Namespace '"/Users/karolkolodziej/repos/highcharts/code/datagrid/datagrid.src"' has no exported member 'Row'.
code/dashboards/es-modules/Dashboards/Plugins/DataGridTypes.d.ts(7,24): error TS2694: Namespace '"/Users/karolkolodziej/repos/highcharts/code/datagrid/datagrid.src"' has no exported member 'Cell'.
code/dashboards/es-modules/Dashboards/Plugins/DataGridTypes.d.ts(9,30): error TS2694: Namespace '"/Users/karolkolodziej/repos/highcharts/code/datagrid/datagrid.src"' has no exported member 'Cell'.
code/dashboards/es-modules/Dashboards/Plugins/DataGridTypes.d.ts(10,33): error TS2694: Namespace '"/Users/karolkolodziej/repos/highcharts/code/datagrid/datagrid.src"' has no exported member 'Options'.
```

File: ts/DataGrid/Cell.ts [hunk 1]
```diff
@@ -113,16 +100,18 @@ class DataGridCell {
...
      * Renders the cell.
      */
     public render(): void {
-        if (!this.column.data) {
-            return;
-        }
-
-        this.setValue(this.column.data[this.row.index], false);
         this.row.htmlElement.appendChild(this.htmlElement);
+        this.setValue(this.column.data?.[this.row.index], false);
```
Comment (ddragula) [inline ts/DataGrid/Cell.ts:114]: I think this should be moved to the `TableCell`.

File: ts/DataGrid/Header/HeaderCell.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            !dataGrid.options?.events?.header?.click
+        ) {
+            return;
+        }
+
+        const onHeaderClick = (): void => {
+            dataGrid.options?.events?.header?.click?.call(
+                column
+            );
+        };
```
Comment (ddragula) [inline ts/DataGrid/Header/HeaderCell.ts:161]: The click event is a common feature of Table and Header cells. Maybe we can move it to an abstract Cell?

File: ts/DataGrid/Cell.ts [hunk 3]
```diff
@@ -58,24 +55,18 @@ class DataGridCell {
...
     /**
      * The row of the cell.
      */
-    public row: DataGridRow;
+    public row: Row;
 
     /**
      * The raw value of the cell.
      */
     public value: DataTable.CellType;
```
Comment (ddragula) [inline ts/DataGrid/Cell.ts:69]: What is `value` for other types of `Cell`s than `TableCell`s? Shouldn't we move this to `TableCell`?

**Review by ddragula — CHANGES_REQUESTED** [review]
Some questions and suggestions for this architecture

**Review by karolkolodziej — APPROVED** [review]

**Review by ddragula — APPROVED** [review]

**Pull request merged**